tests/fv/fv_test.go

```diff
@@ -85,52 +86,112 @@ var _ = Describe("PolicyController", func() {
-			return err
-		}).ShouldNot(HaveOccurred())
-	})
-
-	Context("when etcd data is lost", func() {
+	Context("profiles", func() {
+		var profName string
 		BeforeEach(func() {
-			// Write some data
+			nsName := "peanutbutter"
+			profName = "k8s_ns." + nsName
 			ns := &v1.Namespace{
 				ObjectMeta: metav1.ObjectMeta{
-					Name: "peanutbutter",
+					Name: nsName,
+					Labels: map[string]string{
+						"peanut": "butter",
+					},
 				},
 				Spec: v1.NamespaceSpec{},
 			}
 			_, err := k8sClient.CoreV1().Namespaces().Create(ns)
 			Expect(err).NotTo(HaveOccurred())
 
-			// Wait for it to appear in etcd
 			Eventually(func() *api.Profile {
 				profile, _ := calicoClient.Profiles().Get(api.ProfileMetadata{Name: "k8s_ns.peanutbutter"})
 				return profile
 			}).ShouldNot(BeNil())
 		})
-		It("should recover by rewriting the data to etcd ", func() {
-			Stop(apiserver)
-			err := calicoClient.Profiles().Delete(api.ProfileMetadata{Name: "k8s_ns.peanutbutter"})
+
+		It("should write new profiles in etcd to match namespaces in k8s ", func() {
+			err := calicoClient.Profiles().Delete(api.ProfileMetadata{Name: profName})
+			Expect(err).ShouldNot(HaveOccurred())
+			Eventually(func() error {
+				_, err := calicoClient.Profiles().Get(api.ProfileMetadata{Name: profName})
+				return err
+			}, time.Second*15, 500*time.Millisecond).ShouldNot(HaveOccurred())
+		})
+		It("should update existing profiles in etcd to match namespaces in k8s", func() {
+			profile, err := calicoClient.Profiles().Update(&api.Profile{Metadata: api.ProfileMetadata{Name: profName}})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(profile.Metadata.Labels).To(BeNil())
+			Eventually(func() map[string]string {
+				prof, _ := calicoClient.Profiles().Get(api.ProfileMetadata{Name: profName})
+				return prof.Metadata.Labels
+			}, time.Second*15, 500*time.Millisecond).ShouldNot(BeEmpty())
+		})
+	})
+
+	Context("policies", func() {
+		var policyName string
+		var genPolicyName string
+
+		BeforeEach(func() {
+			// Create a Kubernetes NetworkPolicy.
+			policyName = "jelly"
+			genPolicyName = "knp.default.default." + policyName
+			var np *extensions.NetworkPolicy
+			np = &extensions.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      policyName,
+					Namespace: "default",
+				},
+				Spec: extensions.NetworkPolicySpec{
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"fools": "gold",
+						},
+					},
+				},
+			}
+			err := k8sClient.Extensions().RESTClient().
+				Post().
+				Resource("networkpolicies").
+				Namespace("default").
+				Body(np).
+				Do().Error()
+			Expect(err).NotTo(HaveOccurred())
+
+			// Wait for it to appear in Calico's etcd.
+			Eventually(func() *api.Policy {
+				policy, _ := calicoClient.Policies().Get(api.PolicyMetadata{Name: genPolicyName})
+				return policy
+			}).ShouldNot(BeNil())
+		})
+
+		It("should write new policies in etcd to match policies in k8s", func() {
```

Comment:
wording nit - instead of saying "new policies" it's more clear if this says something like "should re-write policies to etcd to match policies in k8s"